newstitles.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+
+"""
+Newstitles
+---
+
+The task is to write a function that produces a list of article titles from The NY Times (www.nytimes.com) and then
+sorts them by the sum of each title's tokens frequency (let's call this sum `weight` -a token is just a word separated
+by whitespace) among all titles. The output should be this sorted list along with the score for each article title.
+
+For example, if the titles are:
+- Trump goes to Japan
+- Shooting in Texas
+- Japan and North Korea in talks
+- Trump offends Australian people
+- Judges visit Trump
+
+Let's calculate each title's weight, starting with the first one.
+In this case, token `Trump` appears 3 times among all titles. `goes`, 1 time. `to`, 1 time. `Japan`, 2 times.
+Adding up all token frequencies gives us: 3 + 1 + 1 + 2 = 7.
+
+The sorted list would be:
+- Japan and North Korea in talks (8)
```

Comment:
You could have written these as unit tests

Reply:
Agreed, this needs test cases